source/extensions/extensions_build_config.bzl

```diff
@@ -229,7 +229,8 @@ EXTENSIONS = {
     #
     # CacheFilter plugins
     #
-    "envoy.extensions.http.cache.simple":               "//source/extensions/filters/http/cache/simple_http_cache:config",
+    "envoy.extensions.filters.http.cache.file_system_http_cache": "//source/extensions/filters/http/cache/file_system_http_cache:config",
```

Comment:
Why is this in the filter namespace? Isn't this analogous to simple_http_cache and belongs in the `envoy.extensoins.http.cache` tree?

Reply:
There was some discussion about this, and the conclusion as I understood it was that simple http cache was done wrong, because its namespace doesn't correspond to its filesystem location.

I don't know if changing the namespace for simple http cache would make sense (would potentially be a config-breaking change if anyone is using that cache, but seems cleanest), or moving its files to match its namespace and then having this one move correspondingly, or if it's preferred that I just follow suit with the ill-conceived layout now that it's already happened. Or maybe just leaving simple http cache wrong and doing future ones right is best.

Reply:
Can you point to this discussion? I don't see this as a filter, more a plugin to a filter, which is really its own extension type.

Reply:
I prefer the new structure you propose going forward. If we will grow the number of extensions it will pay off. 

We can fix the source layout to be fully consistent across the two. I'm not sure about the API, the package name is material in the protobug type, so the only way we could migrate simple_http_cache is with a copy+paste+deprecate. I'm not sure that is worth it, so I would just leave it there and document for now.

@mattklein123 to sanity check my reasoning here.

Reply:
Sounds good to me. (Though if it was up to me I would just do a breaking change to clean up simple_http_cache, because it's marked WIP [and also terrible], so should be okay to make breaking changes to!)
I'll make that change when I've finished with the test I'm currently writing.

Reply:
The practical advantage of the structure I already had is that it makes it naturally live in the same namespace as all the HttpCache stuff it depends on. *Ideally* all that stuff would *also* be in an Envoy::Extensions::HttpCache namespace. I'm not sure how to strike the balance here - for now I'm moving the paths and api but leaving the mismatched namespace.

Reply:
I'm +1 on the new structure you propose. Given that simple cache is not actually useful in production, IMO it's fine to just break it if you would be willing to do that work. We would definitely appreciate it! Thank you!

Reply:
Yeah, didn't realize it was still WiP. In that case I'm also +1 on the breaking change to clean this up.

Reply:
@ravenblackx are you going to move the simple cache in a separate PR?

Reply:
> @ravenblackx are you going to move the simple cache in a separate PR?

Yes, I'm waiting on that until the other precursor PR that touches it (#23717) is merged so I don't get into horrible merge conflicts with myself.

Reply:
This is done. (#24112)